### PR TITLE
add new cases to cover virsh command:virsh_detach_device_alias

### PR DESCRIPTION
**Pull request by chunfuwen**
virsh_detach_device_alias is used to detach device by its alias name.

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    1) Define the VM with specified serial type device.
+    2) Start the guest and check if start result meets expectation
+    3) Hot unplug serial device and check if meets expectation
+    4) Cold unplug serial device and check if meets expectation
+    5) Shutdown the VM and clean up environment
+    """
+
+    def set_targets(serial):
+        """
+        Set a serial device target attributes.
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: please add docstring for param

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        vm_xml.remove_all_device_by_type('serial')
+        vm_xml.remove_all_device_by_type('console')
+
+        serial_dev = prepare_serial_device()
+        logging.debug('Serial device:\n%s', serial_dev)
+        vm_xml.add_device(serial_dev)
+        vm_xml.sync()
+
+        res = virsh.start(vm_name)
+        libvirt.check_exit_status(res)
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: Better use vm.start()

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: updated

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py (hunk 2 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: updated

Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: updated!

File: libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/domain/virsh_detach_device_alias.py]: updated

**Comment by chunfuwen** [comment]
paste the test result
'''
PASS 1-type_specific.io-github-autotest-libvirt.virsh.detach_serial_device_alias.serial_type_pty.isa-serial
PASS 2-type_specific.io-github-autotest-libvirt.virsh.detach_serial_device_alias.serial_type_pty.pci-serial

PASS 3-type_specific.io-github-autotest-libvirt.virsh.detach_serial_device_alias.serial_type_file.isa-serial
PASS 4-type_specific.io-github-autotest-libvirt.virsh.detach_serial_device_alias.serial_type_file.pci-serial

'''

**Pull request merged**